File: backend/app/models/enums.py
"""Domain enums shared across every other models module.

Pure values — no Pydantic, no FastAPI, no DB. Imported by every other
``models/*.py`` so this file must stay import-cycle-free.
"""
from __future__ import annotations

from enum import Enum


class SupplementCategory(str, Enum):
    """Top-level product categories — maps to the BigQuery ``category`` column.

    Only categories that actually exist in the scraped data are listed; the
    raw↔enum bridge lives in ``services/bigquery.py``. The earlier
    nutrition / endurance / wearables / combat-protection buckets were dropped
    because no scraped product maps to them (they returned empty catalogue
    results).
    """
    STRENGTH_HOME_GYM  = "strength_home_gym"   # raw: gym, general
    TEAM_FOOTBALL      = "team_football"        # raw: football
    TEAM_BASKETBALL    = "team_basketball"      # raw: basketball
    TEAM_VOLLEYBALL    = "team_volleyball"      # raw: Volleyball
    TEAM_RACKET        = "team_racket"          # raw: Racket-Sports
    COMBAT_BOXING_MMA  = "combat_boxing_mma"    # raw: combat-sports


class NutritionSubcategory(str, Enum):
    WHEY_PROTEIN = "whey_protein"
    CREATINE     = "creatine"
    PRE_WORKOUT  = "pre_workout"
    AMINO_ACIDS  = "amino_acids"
    MASS_GAINER  = "mass_gainer"
    VITAMINS     = "vitamins"
    ENERGY_GEL   = "energy_gel"
    ELECTROLYTES = "electrolytes"


class PriceTrend(str, Enum):
    RISING  = "rising"
    FALLING = "falling"
    STABLE  = "stable"


class BrandTier(str, Enum):
    PREMIUM  = "premium"   # Garmin, Nike, Optimum Nutrition
    MID      = "mid"       # Myprotein, Adidas, Everlast
    BUDGET   = "budget"    # generic / unbranded


class AlertType(str, Enum):
    PRICE_DROP    = "price_drop"
    BACK_IN_STOCK = "back_in_stock"
    BUY_SOON      = "buy_soon"
    PRICE_RISE    = "price_rise"


class SortOption(str, Enum):
    SCRAPED_AT_DESC       = "scraped_at_desc"   # newest first (default)
    PRICE_ASC             = "price_asc"
    PRICE_DESC            = "price_desc"
    RATING_DESC           = "rating_desc"
    DISCOUNT_DESC         = "discount_desc"
    PRICE_PER_SERVING_ASC = "price_per_serving_asc"

File: frontend/sportsintelligence/src/app/core/models/enums.ts
/**
 * Domain enums — mirror of `backend/app/models/enums.py`.
 *
 * Modelled as string-literal union types plus `const` value arrays so the
 * frontend can both type-check and iterate (e.g. to render filter options).
 */

// Only categories that exist in the scraped BigQuery data. Mirror of
// backend SupplementCategory enum; the raw↔enum bridge is in
// backend services/bigquery.py. Dropped nutrition/endurance/wearables/
// combat-protection — they returned no products.
export const SUPPLEMENT_CATEGORIES = [
  'strength_home_gym',
  'team_football',
  'team_basketball',
  'team_volleyball',
  'team_racket',
  'combat_boxing_mma',
] as const;
export type SupplementCategory = (typeof SUPPLEMENT_CATEGORIES)[number];

export const NUTRITION_SUBCATEGORIES = [
  'whey_protein',
  'creatine',
  'pre_workout',
  'amino_acids',
  'mass_gainer',
  'vitamins',
  'energy_gel',
  'electrolytes',
] as const;
export type NutritionSubcategory = (typeof NUTRITION_SUBCATEGORIES)[number];

export type PriceTrend = 'rising' | 'falling' | 'stable';

export type BrandTier = 'premium' | 'mid' | 'budget';

export type AlertType = 'price_drop' | 'back_in_stock' | 'buy_soon' | 'price_rise';

export const SORT_OPTIONS = [
  'scraped_at_desc',
  'price_asc',
  'price_desc',
  'rating_desc',
  'discount_desc',
  'price_per_serving_asc',
] as const;
export type SortOption = (typeof SORT_OPTIONS)[number];

/** Human-readable labels for the category enum — used in filter UIs. */
export const CATEGORY_LABELS: Record<SupplementCategory, string> = {
  strength_home_gym: 'Gym',
  team_football: 'Football',
  team_basketball: 'Basketball',
  team_volleyball: 'Volleyball',
  team_racket: 'Racket Sports',
  combat_boxing_mma: 'Combat Sports',
};

export const SORT_LABELS: Record<SortOption, string> = {
  scraped_at_desc: 'Newest first',
  price_asc: 'Lowest price',
  price_desc: 'Highest price',
  rating_desc: 'Highest rated',
  discount_desc: 'Biggest discount',
  price_per_serving_asc: 'Best value / serving',
};
